lib/gem_tracker/ci/github.rb

```diff
@@ -64,15 +64,27 @@ def get_repository
 
   # https://docs.github.com/en/rest/reference/actions#list-jobs-for-a-workflow-run
   def get_run_jobs(jobs_url)
-    jobs_url = "#{jobs_url}?per_page=100"
-    request(jobs_url) do |response|
-      json = JSON.parse(response.body)
-      jobs = json.fetch("jobs") { pp json; raise "Couldn't find runs jobs" }
-      if jobs.size != json['total_count']
-        raise "request did not return all jobs: #{jobs.size} of #{json['total_count']}"
+    total_count = nil
+    page_size = 100
+
+    urls_enum = (1..).lazy.map { |page| "#{jobs_url}?per_page=#{page_size}&page=#{page}" }
+    job_pages_enum = urls_enum.map do |url|
+      request(url) do |response|
+        json = JSON.parse(response.body)
+        jobs = json.fetch("jobs") { pp json; raise "Couldn't find runs jobs" }
+        total_count ||= jobs["total_count"]
+        jobs
       end
-      jobs
     end
+
+    job_pages = job_pages_enum.take_while { |job_page| job_page.size == page_size }.to_a
```

Comment:
This might still do an extra request if the numbers of jobs is a multiple of 100.
Maybe we can keep a counter of how many jobs total have been read?
A more imperative approach might be easier for that.

Reply:
Done, please take a look.